pkg/operator/ceph/disruption/clusterdisruption/pools.go

```diff
@@ -170,30 +190,48 @@ func (r *ReconcileClusterDisruption) reconcileCephFilesystem(cephFilesystemList
 			break
 		}
 		blockOwnerDeletion := false
-		pdb := &policyv1beta1.PodDisruptionBudget{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      pdbName,
-				Namespace: namespace,
-				OwnerReferences: []metav1.OwnerReference{
-					{
-						APIVersion:         filesystem.APIVersion,
-						Kind:               filesystem.Kind,
-						Name:               filesystem.ObjectMeta.Name,
-						UID:                filesystem.UID,
-						BlockOwnerDeletion: &blockOwnerDeletion,
-					},
+		objectMeta := metav1.ObjectMeta{
+			Name:      pdbName,
+			Namespace: namespace,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         filesystem.APIVersion,
+					Kind:               filesystem.Kind,
+					Name:               filesystem.ObjectMeta.Name,
+					UID:                filesystem.UID,
+					BlockOwnerDeletion: &blockOwnerDeletion,
 				},
 			},
+		}
+		usePDBV1, err := mon.UsePDBV1Version(r.context.ClusterdContext.Clientset)
+		if err != nil {
+			return err
+		}
+		if usePDBV1 {
+			pdb := &policyv1.PodDisruptionBudget{
+				ObjectMeta: objectMeta,
+				Spec: policyv1.PodDisruptionBudgetSpec{
+					Selector:     labelSelector,
+					MinAvailable: minAvailable,
+				},
+			}
+			request := types.NamespacedName{Name: pdbName, Namespace: namespace}
+			err := r.reconcileStaticPDB(request, pdb)
+			if err != nil {
+				return errors.Wrapf(err, "failed to reconcile cephfs pdb %v", request)
+			}
+		}
```

Comment:
Looks like you need to `return` before the end of this code block, or else you will create both v1 and v1beta1 pdbs.

Reply:
yes, @travisn,
I should use `continue` as the execution is still inside `for loop`?

Reply:
Done